src/test/java/me/karakelley/http/HttpServerTest.java

```diff
@@ -153,7 +154,7 @@ void testPortTooLarge() throws Exception {
 
   @Test
   void testSocketNotAvailable() {
-    httpServer = new HttpServer(4050);
+    httpServer = new HttpServer(4050, publicDirectoryMock);
```

Comment:
When you have to go back and change every test because you're adding new functionality, it's a good time to take a step back and ask yourself if its really necessary.